### PR TITLE
OCPQE-18089: Config CCO in manual mode and provide credentials manually

### PR DESCRIPTION
**Pull request by jianli-wei**
- add the workflow `cucushift-installer-rehearse-gcp-ipi-cco-manual-users-static` and the provision & deprovision chains
- add the steps `gcp-provision-cco-manual-users-static` & `gcp-deprovision-cco-manual-users-static`, to create/delete the credentials
- add verification test `gcp-ipi-cco-manual-users-static-f28`

### CODE REVIEW

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-gcp-ipi-cco-manual-users-static-f28

File: ci-operator/step-registry/gcp/provision/cco-manual-users/static/gcp-provision-cco-manual-users-static-commands.sh [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  readarray -t roles < <(yq-go r "${creds_requests_dir}/${yaml_filename}" "spec.providerSpec.predefinedRoles" | sed 's/- //g')
+  secret_name=$(yq-go r "${creds_requests_dir}/${yaml_filename}" "spec.secretRef.name")
+  secret_namespace=$(yq-go r "${creds_requests_dir}/${yaml_filename}" "spec.secretRef.namespace")
+  metadata_name=$(yq-go r "${creds_requests_dir}/${yaml_filename}" "metadata.name")
+  if [ -z "$secret_name" ] || [ -z "$secret_namespace" ] || [ -z "$metadata_name" ]; then
+    echo "$(date -u --rfc-3339=seconds) - ERROR: spec.secretRef.name, spec.secretRef.namespace or metadata.name is empty"
+    cat "${creds_requests_dir}/${yaml_filename}"
+    exit 1
+  fi
+  if [ -z "${roles:-}" ]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/gcp/provision/cco-manual-users/static/gcp-provision-cco-manual-users-static-commands.sh:151]: https://github.com/openshift/release/pull/47028#discussion_r1439213632

File: ci-operator/step-registry/gcp/provision/cco-manual-users/static/gcp-provision-cco-manual-users-static-ref.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+ref:
+  as: gcp-provision-cco-manual-users-static
+  from_image:
+    namespace: ocp
+    name: "4.15"
+    tag: upi-installer
+  commands: gcp-provision-cco-manual-users-static-commands.sh
```
Comment (jianlinliu) [inline ci-operator/step-registry/gcp/provision/cco-manual-users/static/gcp-provision-cco-manual-users-static-ref.yaml:7]: https://github.com/openshift/release/pull/47028#discussion_r1439208752

File: ci-operator/step-registry/gcp/provision/cco-manual-users/static/gcp-provision-cco-manual-users-static-commands.sh [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+GCP_SERVICE_ACCOUNT=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${GCP_SERVICE_ACCOUNT}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GOOGLE_PROJECT_ID}"
+fi
+
+# For OCP 4.15+
+CCO_CUSTOM_ROLE="projects/${GOOGLE_PROJECT_ID}/roles/installer_qe_cco_permissions"
+IMAGE_REGISTRY_CUSTOM_ROLE="projects/${GOOGLE_PROJECT_ID}/roles/installer_qe_image_registry_permissions"
```
Comment (jianlinliu) [inline ci-operator/step-registry/gcp/provision/cco-manual-users/static/gcp-provision-cco-manual-users-static-commands.sh:100]: Suggest to add some comments to explain the included permission for the custom roles, and where the permission list gotten from.

File: ci-operator/step-registry/gcp/provision/cco-manual-users/static/gcp-provision-cco-manual-users-static-commands.sh [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+  metadata_name=$(yq-go r "${creds_requests_dir}/${yaml_filename}" "metadata.name")
+  if [ -z "$secret_name" ] || [ -z "$secret_namespace" ] || [ -z "$metadata_name" ]; then
+    echo "$(date -u --rfc-3339=seconds) - ERROR: spec.secretRef.name, spec.secretRef.namespace or metadata.name is empty"
+    cat "${creds_requests_dir}/${yaml_filename}"
+    exit 1
+  fi
+  if [ -z "${roles:-}" ]; then
+    if [[ "${metadata_name}" =~ cloud-credential-operator ]]; then
+      echo "$(date -u --rfc-3339=seconds) - Using pre-configured custom role for cloud-credential-operator"
+      cmd="gcloud iam roles describe $(basename ${CCO_CUSTOM_ROLE}) --project=${GOOGLE_PROJECT_ID}"
```
Comment (jianlinliu) [inline ci-operator/step-registry/gcp/provision/cco-manual-users/static/gcp-provision-cco-manual-users-static-commands.sh:179]: Suggest to introduce `run_command` function:
```
function run_command() {
    local CMD="$1"
    echo "Running Command: ${CMD}"
    eval "${CMD}"
}
```

Comment (jianli-wei) [reply ci-operator/step-registry/gcp/provision/cco-manual-users/static/gcp-provision-cco-manual-users-static-commands.sh:185]: Updated, thanks!

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.14-gcp-ipi-cco-manual-users-static-f28

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-gcp-ipi-cco-manual-users-static-f28

**Pull request merged**